src/components/ProvisionSmartWalletDialog.tsx

```diff
@@ -24,8 +25,13 @@ const useSmartWalletFeeQuery = (rpc?: string) => {
         const feeUnit = params.params.beansPerUnit.find(
           ({ key }: { key: string }) => key === 'feeUnit'
         )?.beans;
+        const feeUnitName = params.params?.feeUnitPrice[0]?.denom;
         assert(feeUnit);
```

Comment:
consider an error message. also they all need to be present (if they aren't the BigInt will make 0n)
```suggestion
assert(beansPerSmartWallet && feeUnit && feeUnitName, 'missing fee params');
```

Reply:
Done